pkgs/default.nix

```diff
@@ -33,6 +33,43 @@
       pretalx-venueless
       pretalx-public-voting
       ;
+
+    libresoc = let
+      libresoc-c4m-jtag = callPackage ./libresoc/c4m-jtag.nix {inherit nmigen nmigen-soc;};
+      libresoc-ieee754fpu = callPackage ./libresoc/ieee754fpu.nix {inherit nmutil nmigen pytest-output-to-files;};
+      libresoc-openpower-isa = callPackage ./libresoc/openpower-isa.nix {
+        inherit
+          libresoc-pyelftools
+          mdis
+          nmigen
+          nmutil
+          pytest-output-to-files
+          ;
+      };
+      libresoc-pyelftools = callPackage ./libresoc/libresoc-pyelftools.nix {};
+      mdis = callPackage ./libresoc/mdis.nix {};
+      nmigen = callPackage ./libresoc/nmigen.nix {};
+      nmigen-soc = callPackage ./libresoc/nmigen-soc.nix {inherit nmigen;};
+      nmutil = callPackage ./libresoc/nmutil.nix {inherit nmigen pytest-output-to-files;};
+      pinmux = callPackage ./libresoc/pinmux.nix {};
+      power-instruction-analyzer = callPackage ./libresoc/power-instruction-analyzer.nix {};
+      pytest-output-to-files = callPackage ./libresoc/pytest-output-to-files.nix {};
+    in rec {
+      # SoC synthesized to nmigen(-soc)
+      soc = callPackage ./libresoc/soc.nix {
+        inherit
+          libresoc-c4m-jtag
+          libresoc-ieee754fpu
+          libresoc-openpower-isa
+          nmigen-soc
+          power-instruction-analyzer
+          pytest-output-to-files
+          ;
+      };
+
+      # SoC synthesized to Verilog
+      verilog = callPackage ./libresoc/verilog.nix {inherit pinmux soc;};
```

Comment:
Discussed with @jleightcap  @albertchae @Chickensoupwithrice

We don't want to build this every time since it's massive, so may as well do something like `soc // { inherit verilog; }` and people can access it as `-A soc.verilog` when needed.